### PR TITLE
Add CI support

### LINKED ISSUE #5: add CI
The projects needs some CI :)

### PR DESCRIPTION
**Pull request by saschagrunert**
This commit adds basic CircleCI support and fixes the related vendoring
and the .gitignore file.

An example pipeline can be found here:
https://circleci.com/workflow-run/e773c6f7-7b57-4530-9049-881dcd99f427

Before this actually works, we have to add CircleCI to this organization (repo).

Closes #5 
Needs #6 
References #7

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@vrothberg @giuseppe @tomsweeneyredhat PTAL

**Comment by saschagrunert** [comment]
I rebased on top of #10 to utilize the go module vendor check.

**Comment by rhatdan** [comment]
LGTM
@vrothberg PTAL

**Comment by saschagrunert** [comment]
The latest workflow can be found here, includes now a vendoring check: https://circleci.com/workflow-run/8713d701-7a25-4fff-b846-0a833bef96ff

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM, but would like a @vrothberg head nod.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  unit-tests:
+    executor: container
+    steps:
+      - <<: *prepare-env
+      - checkout
+      - attach_workspace:
+          at: .
+      - restore_cache:
+          keys:
+            - v1-unit-tests-{{ checksum "vendor.conf" }}
```
Comment (vrothberg) [inline .circleci/config.yml:73]: The `vendor.conf` looks redundant as we're on go modules. Does that have any effect?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      - run:
+          name: unit tests
+          command: make test-unit
+      - save_cache:
+          key: v1-unit-tests-{{ checksum "vendor.conf" }}
```
Comment (vrothberg) [inline .circleci/config.yml:78]: Same here.

**Review by vrothberg — COMMENTED** [review]
LGTM, just one question

File: .circleci/config.yml (hunk 1 above)
Comment (saschagrunert) [reply .circleci/config.yml:73]: Yes, I have to fix this. I'll create a follow-up pr.

**Pull request merged**